blob/service.go

```diff
@@ -245,21 +248,38 @@ func (s *Service) getByCommitment(
 			continue
 		}
 
-		// reconstruct the Blob.
-		blob, err := sharesToBlobs(rawShares[:amount])
+		blob, same, err := constructAndVerifyBlob(rawShares[:amount], commitment)
 		if err != nil {
 			return nil, nil, err
 		}
-
-		// compare commitments.
-		if bytes.Equal(blob[0].Commitment(), commitment) {
-			return blob[0], &proofs, nil
+		if same {
+			return blob, &proofs, nil
 		}
 
 		// drop info of the checked blob
 		rawShares = rawShares[amount:]
+		if len(rawShares) > 0 {
+			// save proof for the last row in case we have rawShares
+			proofs = proofs[len(proofs)-1:]
+		} else {
+			// otherwise slash proofs
```

Comment:
Slash always gives me different vibes in blockchain context :D

Reply:
okie :D